src/Services/Export/WB.Services.Export/Jobs/ExportArchiveHandleService.cs

```diff
@@ -30,19 +31,68 @@ public ExportArchiveHandleService(IFileBasedExportedDataAccessor fileBasedExport
             this.fileNameService = fileNameService;
         }
 
-        public async Task ClearAllExportArchives(TenantInfo tenant)
+        public async Task RunRetentionPolicy(TenantInfo tenant, int? countToKeep, int? daysToKeep)
+        {
+            if(!countToKeep.HasValue && !daysToKeep.HasValue)
+                return;
+            await DeleteExportArchives(tenant, countToKeep, daysToKeep);
+        }
+
+        public async Task ClearExportArchives(TenantInfo tenant)
+        {
+            await DeleteExportArchives(tenant, null, null);
+        }
+        
+        private async Task DeleteExportArchives(TenantInfo tenant, int? countToKeep, int? daysToKeep)
         {
             if (this.externalArtifactsStorage.IsEnabled())
             {
                 var externalStoragePath = this.exportFileAccessor.GetExternalStoragePath(tenant, string.Empty);
                 var items = await this.externalArtifactsStorage.ListAsync(externalStoragePath);
                 if(items == null) return;
 
-                logger.LogInformation("Deleting export archives for tenant: {tenant} - there is {count} files", tenant, items.Count);
+                logger.LogInformation(

```

Comment:
This method contains same removal logic in two parts of code, move logic to separate method.
And cover this new method with tests